control/eiger/eiger_detector.py

```diff
@@ -682,15 +690,25 @@ def initialize_detector(self):
         logging.info("Initializing the detector")
         self._initialize_event.set()
 
-    def fetch_stale_parameters(self):
-        for cfg in self._stale_parameters:
+    def fetch_stale_parameters(self, blocking=False):
+        with self._lock:
+            stale_parameters = set(self._stale_parameters)
+            del self._stale_parameters[:]
+
+        for cfg in stale_parameters:
             if cfg in self.DETECTOR_CONFIG:
                 param = self.read_detector_config(cfg)
             elif cfg in self.STREAM_CONFIG:
                 param = self.read_stream_config(cfg)
             logging.info("Read from detector [{}]: {}".format(cfg, param))
             setattr(self, cfg, param)
-        self._stale_parameters = []
+
+        # _stale_parameters may have been repopulated again by the time we get here and
+        # the flag will stay set - if so we fetch again here or in the next status poll
+        if blocking and self._stale_parameters:
+            logging.info("Stale parameters repopulated - fetching again")
+            self.fetch_stale_parameters(blocking)
+
         self.update_stale_parameters()
 
     def update_stale_parameters(self):
```

Comment:
Here it sets the value in the parameter tree, which presumably mean GETs would receive the correct value. But, at 541 it seems to intercepts the requests for it and manual constructs the response. Maybe they don't do the same thing.